### DISCUSSION TITLE
XCode Installation/Update

### DISCUSSION THREAD

**Discussion by gsomaratne** [CS342/.github#5] [Setup] [answered]
### Description

I have had trouble getting XCode to update. There is no error message to notify me of what went wrong. Each time, after many hours or overnight, it's as if it was never initiated.

### Reproduction

I first tried updating Xcode on the Mac App Store. Then, I deleted my old version of Xcode and tried again. Finally, I tried downloading Xcode from Apple Developer.

### Expected behavior

I thought Xcode would upload overnight, but in the morning it was as if it had never started. I thought the xip file from Apple Developer would lead to a faster download.

### Additional context

2020 MacBook Pro, 2.3 GHz Quad-Core Intel Core i7 Processor, macOS Ventura 13.1 

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
Hi @gsomaratne,
Thank you for your discussion post! As far as I understood, there has yet to be any download progress on your machine, so you would not even get the xip file from the Apple Developer portal. Generally, downloading Xcode from the Developer Portal is a bit more stable than the Mac App Store installation.

A few questions to further troubleshoot:

1. Does it not even start with the download, or does it only progress very slowly?
2. Have you tried a different browser or downloaded it using the university WiFi somewhere on campus? E.g., I get a solid 10 Mbits/s download at the Clark Center on Campus using Stanford WiFi.
3. Do you have enough storage space left on your machine to install Xcode? Generally, more than 30 GB of free would be a good target as you will have the .xip and the extracted version of Xcode on your machine.

**Reply by gsomaratne** [reply]
It is now resolved! Downloading using wifi somewhere else on campus was helpful thank you! It was about the speed of the xip file download

**Reply by PSchmiedmayer** [reply]
@gsomaratne Sounds great, happy to help! Glad to hear that this solved the problem!
Please mark the comment as an answer so we can mark the discussion as solved.
